### ISSUE TITLE
Properly support heredocs in Ruby

### ISSUE THREAD

**Issue by Will-Sommers** [cursorless-dev/cursorless-vscode#593]
There is some discussion in the [PR that added Ruby support](https://github.com/cursorless-dev/cursorless-vscode/pull/578) and I've [opened an issue](https://github.com/tree-sitter/tree-sitter-ruby/issues/204) in the ruby-tree-sitter project.

Currently heredoc behaves mostly like a string, ie. `"take round"` within `<<-DOC (hello) DOC` works. However, `take string` doesn't yet and only selects a portion of the heredoc. I suspect we could do something similar to function used to support `lambda` in Ruby and look for a parent and then check the adjacent sibling.

**Comment by pokey** [comment]
I don't think we can support this one quite as easily as we did with lambda, because from what I can tell the start of the heredoc and the body don't always share a parent.  See for example:

```ruby
a = <<~TEXT
content
TEXT
```

parsed as:

```
(program [0, 0] - [2, 4]
  (assignment [0, 0] - [0, 11]
    left: (identifier [0, 0] - [0, 1])
    right: (heredoc_beginning [0, 4] - [0, 11]))
  (heredoc_body [0, 11] - [2, 4]
    (heredoc_content [0, 11] - [2, 4])
    (heredoc_end [2, 4] - [2, 4])))
```

It's still doable, but would be a fair amount more gymnastics involving descending ancestors of leading siblings etc, and I'd be worried about missing some corner cases

Probably best to push down the route of getting tree-sitter-ruby to support this one as far as we can before we decide pick up the chainsaw 😅
